conan/internal/runner/docker.py

```diff
@@ -0,0 +1,273 @@
+    ConanOutput().status(f'| {msg} |', fg=fg)
+    ConanOutput().status('└'+'─'*(2+len(msg))+'┘\n', fg=fg)
+
+
+class DockerRunner:
+    def __init__(self, conan_api, command, host_profile, build_profile, args, raw_args):
+        import docker
+        import docker.api.build
+        try:
+            self.docker_client = docker.from_env()
+            self.docker_api = docker.APIClient()
+            docker.api.build.process_dockerfile = lambda dockerfile, path: ('Dockerfile', dockerfile)
+        except:
+            raise ConanException("Docker Client failed to initialize."
+                                 "\n - Check if docker is installed and running"
+                                 "\n - Run 'pip install pip install conan[runners]'")
+        self.conan_api = conan_api
+        self.build_profile = build_profile
+        self.args = args
+        self.abs_host_path = make_abs_path(args.path)
+        if args.format:
+            raise ConanException("format argument is forbidden if running in a docker runner")
+
+        # Runner config
+        self.abs_runner_home_path = os.path.join(self.abs_host_path, '.conanrunner')
+        self.abs_docker_path = os.path.join('/root/conanrunner', os.path.basename(self.abs_host_path)).replace("\\","/")
+
+        # Update conan command and some paths to run inside the container
+        raw_args[raw_args.index(args.path)] = self.abs_docker_path
+        self.profiles = []
+        if self.args.profile_build and self.args.profile_host:
+            profile_list = set(self.args.profile_build + self.args.profile_host)
+        else:
+            profile_list = self.args.profile_host or self.args.profile_build
+        
+        # Update the profile paths
+        for i, raw_arg in enumerate(raw_args):
+            for i, raw_profile in enumerate(profile_list):
+                _profile = ProfileLoader.get_profile_path(os.path.join(ConfigAPI(self.conan_api).home(), 'profiles'), raw_profile, os.getcwd())
+                _name = f'{os.path.basename(_profile)}_{i}'
+                if raw_profile in raw_arg:
+                    raw_args[raw_args.index(raw_arg)] = raw_arg.replace(raw_profile, os.path.join(self.abs_docker_path, '.conanrunner/profiles', _name))
+                self.profiles.append([_profile, os.path.join(self.abs_runner_home_path, 'profiles', _name)])
+
+        self.command = ' '.join([f'conan {command}'] + [f'"{raw_arg}"' if ' ' in raw_arg else raw_arg for raw_arg in raw_args] + ['-f json > create.json'])
+
+        # Container config
+        # https://containers.dev/implementors/json_reference/
+        self.configfile = config_parser(host_profile.runner.get('configfile'))
+        self.dockerfile = host_profile.runner.get('dockerfile') or self.configfile.build.dockerfile
+        self.docker_build_context = host_profile.runner.get('build_context') or self.configfile.build.build_context
+        self.image = host_profile.runner.get('image') or self.configfile.image
```

Comment:
I find the preference in selecting from the profile or the config file a bit unnatural, because you can set in the profile contradictory things, like for example using a configfile that points to a dockerfile and be pointing in the profile to another one, shouldn't the configfile and the profile settings that overlap be incompatible in some way?